File: Python/Vehicle.py
#Saya Muhammad Alfi faiz NIM 2207045 mengerjakan
#soal Latihan 2 dalam mata kuliah Desain Pemograman Berorientasi Objek
#untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

class Vehicle:
    def __init__(self, plat_nomor, merk, tahun_produksi, warna):
        # Inisialisasi atribut-atribut kendaraan
        self.__plat_nomor = plat_nomor
        self.__merk = merk
        self.__tahun_produksi = tahun_produksi
        self.__warna = warna

    # Getter methods
    def get_plat_nomor(self):
        # Mendapatkan plat nomor kendaraan
        return self.__plat_nomor

    def get_merk(self):
        # Mendapatkan merk kendaraan
        return self.__merk

    def get_tahun_produksi(self):
        # Mendapatkan tahun produksi kendaraan
        return self.__tahun_produksi

    def get_warna(self):
        # Mendapatkan warna kendaraan
        return self.__warna

    # Setter methods
    def set_plat_nomor(self, plat_nomor):
        # Mengatur plat nomor kendaraan
        self.__plat_nomor = plat_nomor

    def set_merk(self, merk):
        # Mengatur merk kendaraan
        self.__merk = merk

    def set_tahun_produksi(self, tahun_produksi):
        # Mengatur tahun produksi kendaraan
        self.__tahun_produksi = tahun_produksi

    def set_warna(self, warna):
        # Mengatur warna kendaraan
        self.__warna = warna

File: CPP/Vehicle.cpp
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>

using namespace std;

class Vehicle {
  private :

  string plat_nomor;
  string merk;
  string tahun_produksi;
  string warna;

  public :

  Vehicle(){
    
  }

  Vehicle(string plat_nomor){
    this->plat_nomor = plat_nomor;
    merk = "";
    tahun_produksi = "";
    warna = "";
  }

  Vehicle(string plat_nomor, string merk){
    this->plat_nomor = plat_nomor;
    this->merk = merk;
    tahun_produksi = "";
    warna = "";
  }

  Vehicle(string plat_nomor, string merk, string tahun_produksi){
    this->plat_nomor = plat_nomor;
    this->merk = merk;
    this->tahun_produksi = tahun_produksi;
    warna = "";
  }

  Vehicle(string plat_nomor, string merk, string tahun_produksi, string warna){
    this->plat_nomor = plat_nomor;
    this->merk = merk;
    this->tahun_produksi = tahun_produksi;
    this->warna = warna;
  }

  void set_plat_nomor(string plat_nomor)
  {
    this->plat_nomor = plat_nomor;
  }
  void set_merk(string merk)
  {
    this->merk = merk;
  }
  void set_tahun_produksi(string tahun_produksi)
  {
    this->tahun_produksi = tahun_produksi;
  }
  void set_warna(string warna)
  {
    this->warna = warna;
  }

  string get_plat_nomor()
  {
    return plat_nomor;
  }

  string get_merk(){
    return merk;
  }

  string get_tahun_produksi(){
    return tahun_produksi;
  }

  string get_warna(){
    return warna;
  }

  ~Vehicle(){

  }
}